tritium-processor/src/main/java/com/palantir/tritium/processor/TritiumAnnotationProcessor.java

```diff
@@ -0,0 +1,438 @@
+        return SourceVersion.latestSupported();
+    }
+
+    @Override
+    public ImmutableSet<String> getSupportedAnnotationTypes() {
+        return ANNOTATIONS;
+    }
+
+    @Override
+    public boolean process(Set<? extends TypeElement> _annotations, RoundEnvironment roundEnv) {
+        if (roundEnv.processingOver()) {
+            return false;
+        }
+        for (Element element : getElementsToProcess(roundEnv)) {
+            if (element.getKind() != ElementKind.INTERFACE) {
+                messager.printMessage(
+                        Kind.ERROR,
+                        "Only interfaces may be instrumented using @" + TritiumInstrument.class.getSimpleName(),
+                        element);
+                continue;
+            }
+            TypeElement typeElement = (TypeElement) element;
+            List<DeclaredType> allInterfaces = new ArrayList<>();
+            List<DeclaredType> minimalInterfaces = new ArrayList<>();
+            if (getInterfaces(element.asType(), allInterfaces, minimalInterfaces)) {
+                inconsistentElements.add(typeElement.getQualifiedName());
+                continue;
+            }
+            if (allInterfaces.isEmpty()) {
+                messager.printMessage(
+                        Kind.ERROR,
+                        "Cannot generate a instrumented implementation. "
+                                + "The annotated class implements no interfaces.",
+                        typeElement);
+                continue;
+            }
+            try {
+                JavaFile generatedFile = generate(typeElement, allInterfaces, minimalInterfaces);
+                try {
+                    generatedFile.writeTo(filer);
+                } catch (IOException e) {
+                    messager.printMessage(
+                            Kind.ERROR, "Failed to write instrumented class: " + Throwables.getStackTraceAsString(e));
+                }
+            } catch (RuntimeException e) {
+                messager.printMessage(
+                        Kind.ERROR, "Failed to generate instrumented class: " + Throwables.getStackTraceAsString(e));
+            }
+        }
+        return false;
+    }
+
+    private Set<Element> getElementsToProcess(RoundEnvironment env) {
+        Set<Element> currentElements = new HashSet<>(env.getElementsAnnotatedWith(TritiumInstrument.class));
+        for (Name name : inconsistentElements) {
+            currentElements.add(elements.getTypeElement(name));
+        }
+        inconsistentElements.clear();
+        return currentElements;
+    }
+
+    private boolean getInterfaces(
+            List<? extends TypeMirror> mirrors,
+            List<DeclaredType> allInterfaces,
+            @Nullable List<DeclaredType> minimalInterfaces) {
+        boolean error = false;
+        for (TypeMirror mirror : mirrors) {
+            error |= getInterfaces(mirror, allInterfaces, minimalInterfaces);
```

Comment:
Should we short circuit?
```suggestion
            if (isInvalid(mirror, allInterfaces, minimalInterfaces)) {
                return true;
            }
```

Reply:
Good call, updated